eva-accession-release/src/main/java/uk/ac/ebi/eva/accession/release/steps/processors/NamedVariantProcessor.java

```diff
@@ -0,0 +1,107 @@
+
+/**
+ * Transform a named variant {@link VariantType#SEQUENCE_ALTERATION} into a structural variant with symbolic alleles.
+ *
+ * Examples of structural variants taken from
+ * <a href='https://samtools.github.io/hts-specs/VCFv4.3.pdf'>VCFv.3 spec</a> (section 5.3):
+ * <pre>
+ * {@code
+ * #CHROM POS      ID   REF  ALT          QUAL FILTER INFO
+ * 2        321682 .    T    <DEL>        6    PASS   SVTYPE=DEL;END=321887;SVLEN=-205
+ * 2      14477084 .    C    <DEL:ME:ALU> 12   PASS   SVTYPE=DEL;END=14477381;SVLEN=-297
+ * 3       9425916 .    C    <INS:ME:L1>  23   PASS   SVTYPE=INS;END=9425916;SVLEN=6027
+ * }
+ * </pre>
+ *
+ * Note that unlike regular INDELS, variants with symbolic alleles have the context bases only in the REF column.
+ *
+ * Also, note that the deletions have the symbolic allele in the ALT column.
+ */
+public class NamedVariantProcessor implements ItemProcessor<Variant, IVariant> {
+
+    private static Logger logger = LoggerFactory.getLogger(NamedVariantProcessor.class);
+
+    @Override
+    public IVariant process(Variant variant) throws Exception {
+
+        String oldReference = variant.getReference();
+        String oldAlternate = variant.getAlternate();
+
+        String newReference = oldReference;
+        String newAlternate = oldAlternate;
+
+        if (isNamedAllele(oldReference) || isSymbolicAllele(oldReference)) {
+            if (isNamedAllele(oldAlternate) || isSymbolicAllele(oldAlternate)) {
+                throw new IllegalArgumentException(
+                        "This variant (with named/symbolic alleles in both the reference and alternate alleles) can't"
+                        + " be written in VCF, as only the ALT column can have symbolic alleles: "
+                        + variant);
+            } else {
+                // swap the alleles, look this class' documentation
+                newReference = oldAlternate;
+                newAlternate = oldReference;
+            }
+        } else {
+            // normal case without special reference allele: ok as it is
+        }
+
+
+        Variant newVariant = new Variant(variant.getChromosome(), variant.getStart(), variant.getEnd(),
+                                         convertNamedAlleleToSymbolicAllele(newReference),
+                                         convertNamedAlleleToSymbolicAllele(newAlternate));
+
+        newVariant.addSourceEntries(variant.getSourceEntries());
+        newVariant.setMainId(variant.getMainId());
+        return newVariant;
+    }
+
+    /**
+     * Named variants have alleles surrounded by parentheses. Those parentheses will be changed for angular brackets
+     * and white spaces will be replaced by underscore so they can be represented in VCF format as symbolic alleles.
+     */
+    private String convertNamedAlleleToSymbolicAllele(String allele) {
+        if (isNamedAllele(allele)) {
+            return ("<" + removeFirstAndLastCharacters(allele) + ">").replace(" ", "_");
```

Comment:
Please confirm the VCF grammar to make sure other special characters like ";=," are allowed for symbolic alleles

Reply:
```
    alt_id          = ( alnum | (punct - (',' | '<' | '>')) )+ - (punct)+ ;
[...]
    ## Other alternates can be any identifier surrounded by < > symbols
    record_alt_other= ('<' alt_id '>') - ('<DEL>' | '<INS>' | '<DUP>' | '<INV>' | '<CNV>' | '<DUP:TANDEM>' | '<DEL:ME:' (alnum)+ '>' | '<INS:ME:' (alnum)+ '>');
```
I think we have several bugs here:
- the spec doesn't say anything about commas but the validator forbids them (e.g. `<A,T>`)
- the grammar seems to want do `alnum (alnum | some_punct)+` but I tried some strings that should be accepted and they are not.
This is unrelated to this PR, though.